### PR TITLE
Feature/extend backend with ward level boundaries

### PR DESCRIPTION
**Pull request by musamusa**
please refer to #141 for details

### CODE REVIEW

**Comment by musamusa** [comment]
@norkans7 please review

File: ureport/utils/__init__.py [hunk 1]
```diff
@@ -487,20 +489,25 @@ def get_ureporters_locations_stats(org, segment):
 
     location_stats = []
 
-    if not field_type or field_type.lower() not in ['state', 'district']:
+    if not field_type or field_type.lower() not in ['state', 'district', 'ward']:
```
Comment (norkans7) [inline ureport/utils/__init__.py]: Minor, this changes in this method belongs to the frontend PR,

File: ureport/polls/models.py [hunk 2]
```diff
@@ -393,7 +393,7 @@ def fetch_results(self, segment=None):
         from raven.contrib.django.raven_compat.models import client
 
         cache_time = UREPORT_ASYNC_FETCHED_DATA_CACHE_TIME
-        if segment and segment.get('location', "") == "District":
+        if segment and segment.get('location', "") in ("District", "Ward"):
```
Comment (norkans7) [inline ureport/polls/models.py]: minor, this belongs to the frontend PR

File: ureport/polls/models.py [hunk 3]
```diff
@@ -428,7 +428,7 @@ def get_results(self, segment=None):
         if cached_value:
             return cached_value['results']
 
-        if segment and segment.get('location', "") == "District":
+        if segment and segment.get('location', "") in ("District", "Ward"):
```
Comment (norkans7) [inline ureport/polls/models.py]: minor, this belongs to the frontend PR

File: ureport/sql/polls_0039.sql [hunk 4]
```diff
@@ -0,0 +1,111 @@
+--------------------------------------------------------------------------------------
```
Comment (norkans7) [inline ureport/sql/polls_0039.sql]: db triggers has not performed well and we removed it, this is not needed any more instead you need to adjust https://github.com/rapidpro/ureport/blob/master/ureport/polls/models.py#L602 to consider the ward as well

File: ureport/polls/migrations/0039_install_poll_results_count_triggers.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
```
Comment (norkans7) [inline ureport/polls/migrations/0039_install_poll_results_count_triggers.py]: This migrations is not needed any more

**Comment by norkans7** [comment]
Once you fix those, we push this and start to backfill the data to be displayed by next PR

**Comment by musamusa** [comment]
@norkans7 updated

File: ureport/utils/__init__.py [hunk 6]
```diff
@@ -487,20 +489,25 @@ def get_ureporters_locations_stats(org, segment):
...
 
     if field_type == 'state':
-        boundary_top_level = 0 if org.get_config('is_global') else 1
-        boundaries = Boundary.objects.filter(org=org, level=boundary_top_level).values('osm_id', 'name').order_by('osm_id')
+        boundary_top_level = Boundary.COUNTRY_LEVEL if org.get_config('is_global') else Boundary.STATE_LEVEL
+        boundaries = Boundary.objects.filter(org=org, level=boundary_top_level).values('osm_id', 'name')\
+            .order_by('osm_id')
         location_counts = {k[6:]: v for k, v in org_contacts_counts.iteritems() if k.startswith('state')}
-
+    elif field_type == 'ward':
```
Comment (norkans7) [inline ureport/utils/__init__.py]: @musamusa this method is used to display the data on the frontend, so I think these changes for ward support need to be move to the frontend PR, only the constant need to be changes in this PR

**Comment by musamusa** [comment]
@norkans7 updated

**Comment by norkans7** [comment]
Looks good @musamusa, Thanks

**Comment by ribalba** [comment]
Hello @norkans7 I am Didi the CTO of eHA and I just wanted to ask why this is taking quite a lot of time? Is there anything we can do to get this merged faster as we seem to be running behind the master tree?

**Comment by norkans7** [comment]
Hey @ribalba, this PR is ready to go in and it will be included in the next push, I need to check if any org has the ward level location data collected and start backfilling the data from RapidPro

**Pull request merged**